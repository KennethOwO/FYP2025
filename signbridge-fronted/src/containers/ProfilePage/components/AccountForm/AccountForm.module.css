.form {
    display: flex;
    flex-direction: column;
    gap: 25px;
}

.accountFormContainer {
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    transition: box-shadow 0.3s ease-in-out;
    padding: 30px;
    border-radius: 10px;
    background-color: #ffffff;
}

.accountContent h1 {
    font-size: 2rem;
    font-weight: 600;
    margin-bottom: 1rem;
}

.inputGroup {
    display: flex;
    gap: 12px;
}

.label {
    text-align: left;
    margin-bottom: 8px;
}

.radioGroup {
    display: flex;
    gap: 12px;
    width: 100%;
}

.genderLabel {
    margin-left: -85%;
    font-size: 13px;
    color: #2b2b2b;
}

.gender {
    width: 22rem;
    display: flex;
}

.gender input[type="radio"] {
    display: none;
}

/* Style the custom radio buttons */
.gender input[type="radio"] + label {
    cursor: pointer;
    font-size: 16px;
    justify-self: flex-start;
    align-self: flex-start;
}

/* Style for unchecked state */
.gender input[type="radio"] + label:before {
    content: "";
    display: inline-flex;
    width: 10px;
    height: 10px;
    border: 2px solid #000;
    border-radius: 50%;
    margin-right: 10px;
    margin-left: 20px;
}

/* Style for checked state */
.gender input[type="radio"]:checked + label:before {
    background-color: #526b94;
}

.selectContainer {
    width: 100%;
}

.selectTrigger {
    width: 100%;
    background: none;
    padding: 0 0 0 1rem;
    border: none;
    font-size: 0.875rem;
    font-weight: 600;
    box-shadow: inset 0 0 0 2px #efefef;
    border-radius: 12px;
    height: 48px;
    color: #1a1d1f;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.selectTriggerIcon {
    cursor: pointer;
    width: 3rem;
    transition: all 0.3s ease;
}

.selectIconFocus {
    transform: rotate(180deg);
    transition: all 0.3s ease;
}

.selectContent {
    transform: translateY(2px);
}

.selectPopover {
    padding: 0.5rem 0;
    border-radius: 12px;
    box-shadow: inset 0 0 0 2px #efefef;
    background-color: white;
    transition: all 0.3s ease;
    z-index: 99;
}

.selectItem {
    text-align: left;
    padding: 0.5rem 1rem;
    cursor: pointer;
    transition: all 0.3s ease;
    font-size: 0.875rem;
    font-weight: 500;
}

.selectItem:hover {
    background-color: #f5f5f5;
}

.selectItem[data-active-item] {
    background-color: #f5f5f5;
}

.imageUploader {
    display: flex;
    flex-direction: column;
    justify-self: flex-start;
    align-self: flex-start;
    margin-top: 20px;
}

.uploadedImage {
    cursor: pointer;
    border-radius: 50%;
    width: 100px;
    height: 100px;
    object-fit: cover;
    object-position: center;
}

.uploadButton {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-top: 0.25rem;
    height: 48px;
    font-size: 15px;
    color: #000000;
    background-color: rgb(255, 255, 255);
    border: black 1px solid;
    border-radius: 8px;
    cursor: pointer;
    transition: all 0.3s ease;
}

.uploadButton:hover {
    color: #ffffff;
    background-color: #000000;
}

.submitButton {
    margin-top: 1rem;
    justify-self: flex-end;
    align-self: flex-end;
    padding-left: 20px;
    padding-right: 20px;
    height: 48px;
    font-size: 15px;
    color: #fcfcfc;
    background-color: #1a1d1f;
    border-radius: 12px;
    border: none;
    cursor: pointer;
}

@media screen and (max-width: 1100px) {
    .accountContent h1 {
        font-size: 1.5rem;
    }

    .genderLabel {
        margin-left: -85%;
    }

    .gender {
        width: 12rem;
    }

    .label {
        font-size: 0.875rem;
    }
}

@media screen and (max-width: 768px) {
    .accountFormContainer {
        padding: 20px;
    }

    .accountContent h1 {
        font-size: 1.5rem;
    }

    .genderLabel {
        margin-left: -60%;
        font-size: 14px;
    }

    .gender {
        width: 9rem;
        margin-top: 15px;
    }

    .gender input[type="radio"] + label {
        font-size: 12px;
    }

    .gender input[type="radio"] + label:before {
        width: 8px;
        height: 8px;
        margin-right: 8px;
        margin-left: 16px;
    }
}
