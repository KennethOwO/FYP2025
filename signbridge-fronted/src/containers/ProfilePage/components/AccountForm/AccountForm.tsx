import React, { ChangeEvent, useState, useEffect } from "react";
import * as Ariakit from "@ariakit/react";
import { ChevronDownIcon } from "@radix-ui/react-icons";
import toast from "react-hot-toast";
import AccountInputField from "../accountInputFields/accountInputFields";
import { useTranslation } from "react-i18next";
import style from "./AccountForm.module.css";
// @ts-ignore
import { storage } from "../../../../../firebase";
import { ref, uploadBytes, getDownloadURL } from "firebase/storage";
import { FetchAllCountries } from "@root/services/account.service";
import { UpdateUserProfileById } from "@root/services/account.service";

import { useUserStore } from "@root/store/userStore";
import { getAuth } from "firebase/auth";

type Country = {
  country: string;
  state: State[];
};

type State = {
  name: string;
  city: string[];
};

const AccountForm = () => {
  const { user, setUser } = useUserStore();
  const [formData, setFormData] = useState({
    username: user?.username ?? "",
    firstName: user?.first_name ?? "",
    lastName: user?.last_name ?? "",
    user_id: user?.user_id ?? "",
    age: user?.age ?? "",
    gender: user?.gender ?? "",
    race: user?.race ?? "",
    country: user?.country ?? "",
    city: user?.city ?? "",
    state: user?.state ?? "",
    picture: null as File | null,
  });

  const [error, setError] = useState({
    username: "",
    firstName: "",
    lastName: "",
    email: "",
    age: "",
    gender: "",
    race: "",
  });
  const currentUser = getAuth().currentUser;

  const { t } = useTranslation();
  const [countries, setCountries] = useState<Country[] | null>(null);
  const [selectCountryOpen, setSelectCountryOpen] = useState(false);
  const [selectCityOpen, setSelectCityOpen] = useState(false);
  const [selectStateOpen, setSelectStateOpen] = useState(false);

  const [selectedCountryOption, setSelectedCountryOption] = useState(
    user?.country ?? ""
  );
  const [selectedStateOption, setSelectedStateOption] = useState(
    user?.state ?? ""
  );
  const [selectedCityOption, setSelectedCityOption] = useState(
    user?.city ?? ""
  );

  const [image, setImage] = useState<string | null>(user?.picture ?? null);
  const [customKey, setCustomKey] = useState(0);

  async function fetchCountry() {
    if (currentUser) {
      const response = await FetchAllCountries(currentUser);
      setCountries(response.data);
    }
  }

  useEffect(() => {
    fetchCountry();
  }, []);

  const handleChange = (
    e: ChangeEvent<HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>
  ) => {
    const { name, value, type } = e.target;
    let isValid = true;

    if (isValid) {
      if (type === "radio" && name === "gender") {
        setFormData({ ...formData, gender: value });
      } else if (type === "radio") {
        setFormData({ ...formData, [name]: parseInt(value) });
      } else {
        setFormData({ ...formData, [name]: value });
      }
    }

    let errorMessage = "";

    switch (name) {
      case "firstName":
        errorMessage = validateFirstName(value)
          ? ""
          : "First name must be at least 3 characters long";
        break;
      case "lastName":
        errorMessage = validateLastName(value)
          ? ""
          : "Last name must be at least 3 characters long";
        break;
      case "age":
        errorMessage = validateAge(value)
          ? ""
          : "Age must be between 2 and 150";
        break;
      case "race":
        errorMessage = validateRace(value)
          ? ""
          : "Race must contain only letters and be at least 3 characters long";
        break;
      default:
        break;
    }

    setError((prevError) => ({
      ...prevError,
      [name]: errorMessage,
    }));

    setFormData((prevFormData) => ({
      ...prevFormData,
      [name]: value,
    }));
  };

  const handleImageChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const file = e.target.files && e.target.files[0];

    setFormData((prevFormData) => ({
      ...prevFormData,
      picture: e.target.files ? e.target.files[0] : null,
    }));

    if (file) {
      const reader = new FileReader();

      reader.onloadend = () => {
        if (typeof reader.result === "string") {
          setImage(reader.result);
        }
      };
      reader.readAsDataURL(file);
    }
  };

  // ---------- Validation Functions ----------
  const validateFirstName = (value: string) => {
    if (value && value.trim() !== "" && value.length < 3) {
      setError((prev) => ({ ...prev, firstName: t("firstNameLength") }));
      return false;
    }
    setError((prev) => ({ ...prev, firstName: "" }));
    return true;
  };

  const validateLastName = (value: string) => {
    if (value && value.trim() !== "" && value.length < 3) {
      setError((prev) => ({ ...prev, lastName: t("lastNameLength") }));
      return false;
    }
    setError((prev) => ({ ...prev, lastName: "" }));
    return true;
  };

  const validateAge = (value: string) => {
    if (
      value &&
      value.trim() !== "" &&
      (+value < 2 || +value > 150 || isNaN(+value))
    ) {
      setError((prev) => ({ ...prev, age: t("ageLength") }));
      return false;
    }
    setError((prev) => ({ ...prev, age: "" }));
    return true;
  };

  const validateRace = (value: string) => {
    if (value && value.trim() !== "") {
      const lettersRegex = /^[a-zA-Z]+$/;
      if (!lettersRegex.test(value)) {
        setError((prev) => ({ ...prev, race: t("raceLetters") }));
        return false;
      } else if (value.length < 3) {
        setError((prev) => ({ ...prev, race: t("raceLength") }));
        return false;
      }
    }
    setError((prev) => ({ ...prev, race: "" }));
    return true;
  };

  const uploadInputData = async (url?: string) => {
    if (!user) {
      return;
    }

    const data = new FormData();

    if (formData.username) {
      data.append("username", formData.username);
    }

    if (formData.picture) {
      data.append("picture", formData.picture);
    }

    if (formData.firstName) {
      data.append("first_name", formData.firstName);
    }

    if (formData.lastName) {
      data.append("last_name", formData.lastName);
    }

    if (formData.age) {
      data.append("age", formData.age);
    }

    if (formData.gender) {
      data.append("gender", formData.gender);
    }

    if (formData.race) {
      data.append("race", formData.race);
    }

    if (formData.country) {
      data.append("country", formData.country);
    }

    if (formData.city) {
      data.append("city", formData.city);
    }

    if (formData.state) {
      data.append("state", formData.state);
    }

    if (url) {
      data.append("picture", url);
    }
    if (currentUser) {
      const result = await UpdateUserProfileById(
        user.firebase_id,
        data,
        currentUser
      );
      if (result) {
        setUser({
          ...user,
          username: formData.username,
          first_name: formData.firstName,
          last_name: formData.lastName,
          age: formData.age,
          gender: formData.gender,
          race: formData.race,
          country: formData.country,
          state: formData.state,
          city: formData.city,
          picture: url ?? user.picture,
        });
      }
    }
  };

  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const isFirstNameValid = validateFirstName(formData.firstName);
    const isLastNameValid = validateLastName(formData.lastName);
    const isAgeValid = validateAge(formData.age ?? "");
    const isRaceValid = validateRace(formData.race);
    if (isFirstNameValid && isLastNameValid && isAgeValid && isRaceValid) {
      if (user && formData.picture) {
        const storageRef = ref(storage, `userProfileImage/${user.firebase_id}`);
        await uploadBytes(storageRef, formData.picture)
          .then(async (snapshot) => {
            const downloadURL = await getDownloadURL(storageRef);
            await uploadInputData(downloadURL);
          })
          .catch((error) => {
            // console.log(error);
          });
      } else {
        await uploadInputData();
      }

      toast.success(t("personalSuccess"));
    } else {
      toast.error(t("personalFailed"));
    }
  };

  return (
    <div className={style.accountFormContainer}>
      <div className={style.accountContent}>
        <h1>{t("personalDetail")}</h1>
        <form onSubmit={handleSubmit} className={style.form} key={customKey}>
          <AccountInputField
            label={t("accountUsername")}
            name="username"
            value={formData.username}
            onChange={handleChange}
            error={error.username}
          />
          <div className={style.inputGroup}>
            <AccountInputField
              label={t("accountFirstName")}
              name="firstName"
              value={formData.firstName}
              onChange={handleChange}
              error={error.firstName}
            />
            <AccountInputField
              label={t("accountLastName")}
              name="lastName"
              value={formData.lastName}
              onChange={handleChange}
              error={error.lastName}
            />
          </div>
          <div className={style.inputGroup}>
            <AccountInputField
              label={t("accountAge")}
              name="age"
              value={formData.age}
              onChange={handleChange}
              error={error.age}
            />
            <div className={style.radio_group}>
              <label className={style.genderLabel}>{t("accountGender")}</label>
              <div className={style.gender}>
                <input
                  type="radio"
                  name="gender"
                  id="male"
                  value="male"
                  onChange={handleChange}
                  checked={formData.gender === "male"}
                />
                <label htmlFor="male"> {t("accountMale")}</label>
                <input
                  type="radio"
                  name="gender"
                  id="female"
                  value="female"
                  className="female"
                  onChange={handleChange}
                  checked={formData.gender === "female"}
                />
                <label htmlFor="female"> {t("accountFemale")}</label>
              </div>
            </div>
          </div>
          <AccountInputField
            label={t("accountRace")}
            name="race"
            value={formData.race}
            onChange={handleChange}
            error={error.race}
          />

          <div className={style.selectContainer}>
            <Ariakit.SelectProvider
              setOpen={(open) => {
                setSelectCountryOpen(open);
              }}
              setValue={(value) => {
                setSelectedCountryOption(value);
              }}
              defaultValue={formData.country ?? ""}
            >
              <Ariakit.SelectLabel className={style.label}>
                {t("accountCountry")}
              </Ariakit.SelectLabel>
              <Ariakit.Select className={style.selectTrigger}>
                {selectedCountryOption
                  ? selectedCountryOption
                  : t("selectCountry")}
                <div
                  className={`${style.selectTriggerIcon} ${
                    selectCountryOpen ? style.selectIconFocus : ""
                  }`}
                >
                  <ChevronDownIcon />
                </div>
              </Ariakit.Select>
              <Ariakit.SelectPopover
                gutter={4}
                sameWidth
                className={style.selectPopover}
              >
                {countries?.map((country, index) => (
                  <Ariakit.SelectItem
                    key={index}
                    className={style.selectItem}
                    value={country.country}
                    onClick={() => {
                      setSelectedCountryOption(country.country);
                      setFormData((prevFormData) => ({
                        ...prevFormData,
                        country: country.country,
                      }));
                      setSelectCountryOpen(false);
                    }}
                  />
                ))}
              </Ariakit.SelectPopover>
            </Ariakit.SelectProvider>
          </div>

          <div className={style.selectContainer}>
            <Ariakit.SelectProvider
              setOpen={(open) => {
                setSelectStateOpen(open);
              }}
              setValue={(value) => {
                setSelectedStateOption(value);
              }}
              defaultValue={formData.state ?? ""}
            >
              <Ariakit.SelectLabel className={style.label}>
                {t("accountState")}
              </Ariakit.SelectLabel>
              <Ariakit.Select className={style.selectTrigger}>
                {selectedStateOption ? selectedStateOption : t("selectState")}
                <div
                  className={`${style.selectTriggerIcon} ${
                    selectStateOpen ? style.selectIconFocus : ""
                  }`}
                >
                  <ChevronDownIcon />
                </div>
              </Ariakit.Select>
              <Ariakit.SelectPopover
                gutter={4}
                sameWidth
                className={style.selectPopover}
              >
                {selectedCountryOption
                  ? countries
                      ?.filter(
                        (country) => country.country === selectedCountryOption
                      )[0]
                      .state.map((state, index) => (
                        <Ariakit.SelectItem
                          key={index}
                          className={style.selectItem}
                          value={state.name}
                          onClick={() => {
                            setSelectedStateOption(state.name);
                            setFormData((prevFormData) => ({
                              ...prevFormData,
                              state: state.name,
                            }));
                            setSelectStateOpen(false);
                            setSelectedCityOption("");
                          }}
                        />
                      ))
                  : null}
              </Ariakit.SelectPopover>
            </Ariakit.SelectProvider>
          </div>

          <div className={style.selectContainer}>
            <Ariakit.SelectProvider
              setOpen={(open) => {
                setSelectCityOpen(open);
              }}
              setValue={(value) => {
                setSelectedCityOption(value);
              }}
              defaultValue={formData.city ?? ""}
            >
              <Ariakit.SelectLabel className={style.label}>
                {t("accountCity")}
              </Ariakit.SelectLabel>
              <Ariakit.Select className={style.selectTrigger}>
                {selectedCityOption ? selectedCityOption : t("selectCity")}
                <div
                  className={`${style.selectTriggerIcon} ${
                    selectCityOpen ? style.selectIconFocus : ""
                  }`}
                >
                  <ChevronDownIcon />
                </div>
              </Ariakit.Select>
              <Ariakit.SelectPopover
                gutter={4}
                sameWidth
                className={style.selectPopover}
              >
                {selectedStateOption
                  ? countries
                      ?.filter(
                        (country) => country.country === selectedCountryOption
                      )[0]
                      .state.filter(
                        (state) => state.name === selectedStateOption
                      )[0]
                      .city.map((city, index) => (
                        <Ariakit.SelectItem
                          key={index}
                          className={style.selectItem}
                          value={city}
                          onClick={() => {
                            setSelectedCityOption(city);
                            setFormData((prevFormData) => ({
                              ...prevFormData,
                              city: city,
                            }));
                            setSelectCityOpen(false);
                          }}
                        />
                      ))
                  : null}
              </Ariakit.SelectPopover>
            </Ariakit.SelectProvider>
          </div>

          {/* Upload Image */}
          <div className={style.imageUploader}>
            <label htmlFor="file" className={style.uploadLabel}>
              {image ? (
                <img
                  src={image ?? ""}
                  alt="Uploaded"
                  className={style.uploadedImage}
                />
              ) : (
                <span>{t("uploadAccImage")}</span>
              )}
            </label>
            <label htmlFor="file" className={style.uploadButton}>
              {t("selectAccImage")}
              <input
                type="file"
                id="file"
                className={style.uploadInput}
                onChange={handleImageChange}
                style={{ display: "none" }}
                accept="image/jpeg, image/png, image/jpg"
              />
            </label>
          </div>

          <button type="submit" className={style.submitButton}>
            {t("submit_btn")}
          </button>
        </form>
      </div>
    </div>
  );
};

export default AccountForm;
