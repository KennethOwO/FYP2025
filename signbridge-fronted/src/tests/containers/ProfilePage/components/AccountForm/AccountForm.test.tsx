import React from "react";
import AccountForm from "@root/containers/ProfilePage/components/AccountForm/AccountForm";
import { create } from "react-test-renderer";
import { cleanup } from "@testing-library/react";

// Mock the translation functions
jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock the translation function to return the key
    i18n: {
      changeLanguage: jest.fn(), // Mock the changeLanguage function
    },
  }),
}));
// Define the type of the mocked function
jest.mock("js-cookie", () => ({
  get: jest.fn(), // No need to specify return type here
}));
describe("Test AccountForm", () => {
  afterEach(() => {
    // Clean up after each test
    cleanup();
    // Reset the cookie mock to avoid affecting other tests
    Object.defineProperty(document, "cookie", {
      writable: true,
      value: "",
    });
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    // call the set method of Cookies
    const tree = create(<AccountForm />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
