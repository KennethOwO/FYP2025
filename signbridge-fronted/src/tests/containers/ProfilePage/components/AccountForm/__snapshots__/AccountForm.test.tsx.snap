// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AccountForm should render correctly 1`] = `
<div>
  <div>
    <h1>
      personalDetail
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="undefined "
      >
        <label
          className="undefined "
        >
          accountUsername
        </label>
        <input
          className="undefined "
          name="username"
          onChange={[Function]}
          placeholder=" "
          type="text"
          value=""
        />
        <div
          className="undefined "
        />
      </div>
      <div>
        <div
          className="undefined "
        >
          <label
            className="undefined "
          >
            accountFirstName
          </label>
          <input
            className="undefined "
            name="firstName"
            onChange={[Function]}
            placeholder=" "
            type="text"
            value=""
          />
          <div
            className="undefined "
          />
        </div>
        <div
          className="undefined "
        >
          <label
            className="undefined "
          >
            accountLastName
          </label>
          <input
            className="undefined "
            name="lastName"
            onChange={[Function]}
            placeholder=" "
            type="text"
            value=""
          />
          <div
            className="undefined "
          />
        </div>
      </div>
      <div>
        <div
          className="undefined "
        >
          <label
            className="undefined "
          >
            accountAge
          </label>
          <input
            className="undefined "
            name="age"
            onChange={[Function]}
            placeholder=" "
            type="text"
            value=""
          />
          <div
            className="undefined "
          />
        </div>
        <div>
          <label>
            accountGender
          </label>
          <div>
            <input
              checked={false}
              id="male"
              name="gender"
              onChange={[Function]}
              type="radio"
              value="male"
            />
            <label
              htmlFor="male"
            >
               
              accountMale
            </label>
            <input
              checked={false}
              className="female"
              id="female"
              name="gender"
              onChange={[Function]}
              type="radio"
              value="female"
            />
            <label
              htmlFor="female"
            >
               
              accountFemale
            </label>
          </div>
        </div>
      </div>
      <div
        className="undefined "
      >
        <label
          className="undefined "
        >
          accountRace
        </label>
        <input
          className="undefined "
          name="race"
          onChange={[Function]}
          placeholder=" "
          type="text"
          value=""
        />
        <div
          className="undefined "
        />
      </div>
      <div>
        <div
          id=":r0:"
          onClick={[Function]}
          style={
            {
              "cursor": "default",
            }
          }
        >
          accountCountry
        </div>
        <button
          aria-autocomplete="none"
          aria-expanded={false}
          aria-haspopup="listbox"
          onBlur={[Function]}
          onClick={[Function]}
          onClickCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onKeyDownCapture={[Function]}
          onKeyPressCapture={[Function]}
          onKeyUp={[Function]}
          onLoadedMetadataCapture={[Function]}
          onMouseDown={[Function]}
          onMouseDownCapture={[Function]}
          role="combobox"
        >
          selectCountry
          <div
            className="undefined "
          >
            <svg
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </button>
        <div
          style={
            {
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": "max-content",
            }
          }
        >
          <div
            data-dialog=""
            data-placing={true}
            hidden={true}
            id=":r1:"
            onBlur={[Function]}
            onBlurCapture={[Function]}
            onClickCapture={[Function]}
            onFocus={[Function]}
            onFocusCapture={[Function]}
            onKeyDown={[Function]}
            onKeyDownCapture={[Function]}
            onKeyPressCapture={[Function]}
            onKeyUpCapture={[Function]}
            onMouseDown={[Function]}
            onMouseDownCapture={[Function]}
            role="listbox"
            style={
              {
                "display": "none",
                "position": "relative",
              }
            }
            tabIndex={-1}
          />
        </div>
      </div>
      <div>
        <div
          id=":r4:"
          onClick={[Function]}
          style={
            {
              "cursor": "default",
            }
          }
        >
          accountState
        </div>
        <button
          aria-autocomplete="none"
          aria-expanded={false}
          aria-haspopup="listbox"
          onBlur={[Function]}
          onClick={[Function]}
          onClickCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onKeyDownCapture={[Function]}
          onKeyPressCapture={[Function]}
          onKeyUp={[Function]}
          onLoadedMetadataCapture={[Function]}
          onMouseDown={[Function]}
          onMouseDownCapture={[Function]}
          role="combobox"
        >
          selectState
          <div
            className="undefined "
          >
            <svg
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </button>
        <div
          style={
            {
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": "max-content",
            }
          }
        >
          <div
            data-dialog=""
            data-placing={true}
            hidden={true}
            id=":r5:"
            onBlur={[Function]}
            onBlurCapture={[Function]}
            onClickCapture={[Function]}
            onFocus={[Function]}
            onFocusCapture={[Function]}
            onKeyDown={[Function]}
            onKeyDownCapture={[Function]}
            onKeyPressCapture={[Function]}
            onKeyUpCapture={[Function]}
            onMouseDown={[Function]}
            onMouseDownCapture={[Function]}
            role="listbox"
            style={
              {
                "display": "none",
                "position": "relative",
              }
            }
            tabIndex={-1}
          />
        </div>
      </div>
      <div>
        <div
          id=":r8:"
          onClick={[Function]}
          style={
            {
              "cursor": "default",
            }
          }
        >
          accountCity
        </div>
        <button
          aria-autocomplete="none"
          aria-expanded={false}
          aria-haspopup="listbox"
          onBlur={[Function]}
          onClick={[Function]}
          onClickCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onKeyDownCapture={[Function]}
          onKeyPressCapture={[Function]}
          onKeyUp={[Function]}
          onLoadedMetadataCapture={[Function]}
          onMouseDown={[Function]}
          onMouseDownCapture={[Function]}
          role="combobox"
        >
          selectCity
          <div
            className="undefined "
          >
            <svg
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </button>
        <div
          style={
            {
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": "max-content",
            }
          }
        >
          <div
            data-dialog=""
            data-placing={true}
            hidden={true}
            id=":r9:"
            onBlur={[Function]}
            onBlurCapture={[Function]}
            onClickCapture={[Function]}
            onFocus={[Function]}
            onFocusCapture={[Function]}
            onKeyDown={[Function]}
            onKeyDownCapture={[Function]}
            onKeyPressCapture={[Function]}
            onKeyUpCapture={[Function]}
            onMouseDown={[Function]}
            onMouseDownCapture={[Function]}
            role="listbox"
            style={
              {
                "display": "none",
                "position": "relative",
              }
            }
            tabIndex={-1}
          />
        </div>
      </div>
      <div>
        <label
          htmlFor="file"
        >
          <span>
            uploadAccImage
          </span>
        </label>
        <label
          htmlFor="file"
        >
          selectAccImage
          <input
            accept="image/jpeg, image/png, image/jpg"
            id="file"
            onChange={[Function]}
            style={
              {
                "display": "none",
              }
            }
            type="file"
          />
        </label>
      </div>
      <button
        type="submit"
      >
        submit_btn
      </button>
    </form>
  </div>
</div>
`;
